package.json

```diff
@@ -62,6 +62,7 @@
     "libp2p-crypto": "^0.17.5",
     "multiaddr": "^7.4.2",
     "peer-id": "^0.13.11",
+    "public-ip": "^4.0.2",
```

Comment:
I've got some heartburn about using this dependency to determine our IP.
I think we should instead just add some config to turn off the current IP voting mechanism. Then we might better control it on the lodestar level as needed.

We might also look at what lighthouse is doing here. This IP voting behavior was lifted from their discv5 implementation.

Reply:
yeah determining the public ip thru a third party like this is not perfect, I used it just because there is no api to know the public ip.
> I think we should instead just add some config to turn off the current IP voting mechanism. Then we might better control it on the lodestar level as needed.

currently we create Discv5 instance through libp2p so I can't think of how to pass additional config to it

> We might also look at what lighthouse is doing here. This IP voting behavior was lifted from their discv5 implementation.

it's good to know. I checked this in the spec but seems like it's not part of it.. Maybe we can look at prysm as well...

Reply:
So what I see is prysm always get external ip address at startup and set to its local enr https://github.com/prysmaticlabs/prysm/blob/master/beacon-chain/p2p/service.go#L182 so it'll never have this isslue

Reply:
@wemeetagain can you double check with lighthouse, I'm not familiar with it :D

Reply:
one possible solution would be to check in PONG message, if the voted ip address is part of network interfaces, we'll continue. Otherwise just ignore.

It means that we'll only insert multiaddrs to ENR if it's a public server (where we have a static public/external ip address). For pc/laptop environment connected to LAN, we'll not insert it.

Reply:
had a discussion with @wemeetagain . The final conclusion is to check if public ip address in local network interface, if not just wipe it at lodestar bootup time, not at discv5.